server/monitor.go

```diff
@@ -32,17 +35,18 @@ type Monitor struct {
 	ival time.Duration
 }
 
-// Monitor ensures a Heart is still beating until a channel is closed, returning
-// an error if the heartbeats fail.
-func (m *Monitor) Monitor(hrt Heart, stop chan bool) error {
+// Monitor periodically checks the given Heart to make sure it
+// beats successfully. If the heartbeat check fails for any
+// reason, an error is returned. If the supplied channel is
+// closed, Monitor returns ErrShutdown.
+func (m *Monitor) Monitor(hrt heart.Heart, sdc <-chan bool) error {
```

Comment:
If the values in sdc don't matter, a channel of `struct{}` would be more appropriate.